analytics/src/main/java/com/segment/analytics/internal/AnalyticsClient.java

```diff
@@ -120,14 +121,18 @@ public void run() {
   }
 
   public int messageSizeInBytes(Message message) {
+    //@jorgen25 test idea: Check if this is correctly testing byte size and not just char length with regular characters
     Gson gson = new Gson();
     String stringifiedMessage = gson.toJson(message);
-    return stringifiedMessage.length();
+
+    int sizeInBytes = stringifiedMessage.getBytes(StandardCharsets.UTF_8).length;
```

Comment:
This needs to be the size of the message _plus_ the size of the comma which will come after the message in the JSON _plus_ (outside of this method) the size of the `batch:[` and `]` characters, otherwise you risk exceeding the total message size limit.

Reply:
@CraigStuntzWillowTree  yeah you are right i remember you mentioned that in other PRs i will consider that as well

Reply:
Hello @CraigStuntzWillowTree, can you please review whenever you get a chance? I made some changes  to the library to fix the size issue
https://github.com/segmentio/analytics-java/pull/298/commits/bf038aa00e468c5c112df0b4b48711ae48aa0818

Thanks!